Imgix/IGXClient.m

```diff
@@ -64,12 +64,20 @@ - (NSString *)queryStringWithPath:(NSString *)path {
 	}];
 
 	NSString *query = [components componentsJoinedByString:@"&"];
+    if (query.length > 0) {
+        query = [@"?" stringByAppendingString:query];
+    }
 
 	if (self.token) {
-		NSString *input = [[NSString alloc] initWithFormat:@"%@%@?%@", self.token, path, query];
+		NSString *input = [[NSString alloc] initWithFormat:@"%@%@%@", self.token, path, query];
 		NSString *signature = [input imgix_MD5];
+        
 		[components addObject:[[NSString alloc] initWithFormat:@"s=%@", signature]];
```

Comment:
While this isn't you're change, there is a memory leak here. It should be either a value that is released after adding:

```objc
NSString *sig = [[NSString alloc] initWithFormat:@"s=%@", signature];
[components addObject:sig];
[sig release];
```
 or an autoreleased string:

```objc
[components addObject:[NSString stringWithFormat:@"s=%@", signature]];
```